test/pylib/cpp/unit/unit_facade.py

```diff
@@ -33,17 +38,33 @@ def run_test(
         test_args: Sequence[str] = (),
         env: dict = None,
     ) -> tuple[list[CppTestFailure], str] | tuple[None, str]:
+        root_log_dir = self.temp_dir / mode
+        stdout_file_path = root_log_dir / f"{test_name}_stdout.log"
+        test = make_test_object(test_name, file_name.parent.name, mode, original_name, log_dir=self.temp_dir)
+
+        resource_gather = get_resource_gather(self.gather_metrics, test=test)
```

Comment:
change in unit_facade and boost_facade looks quite similar. Did you try to implement some common method?

Reply:
Taking a look one more time on this, I think I know how to extract some common logic to the `CppTestFacade`. I'll update it.